plugin/src/main/java/org/wildfly/plugin/provision/DeploymentResolution.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.plugin.provision;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Resolve possible deployments from the Maven project dependencies.
+ */
+class DeploymentResolution {
+
+    private static final String WAR = "war";
+    private static final String EAR = "ear";
+    private static final String RAR = "rar";
+    private static final String JAR = "jar";
+    private static final Set<String> SUPPORTED_DEPLOYMENT_TYPES = Set.of(EAR, JAR, RAR, WAR);
+    private static final String TEST_JAR = "test-jar";
+
+    static DeploymentResolution getInstance(MavenProject project) throws MojoExecutionException {
+        return new DeploymentResolution(project);
+    }
+
+    private final Map<String, File> files = new TreeMap<>();
```

Comment:
What is the reason for a `TreeMap`?  It seems we could just use a `HashMap` which would perform a little better.

Reply:
A letfover... Good catch.